### PR TITLE
Introduce `yourbones_ppx` for literal `tez` values

### PR DESCRIPTION
**Pull request by mspwn**
This is my first ppx and I'm still a newbie.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,6 +11,7 @@ build:
 test:
 	dune runtest --no-buffer -j 1
 
+
```
Comment (xvw) [inline Makefile]: Could you remove that line?

File: dune-project [hunk 2]
```diff
@@ -24,3 +25,15 @@
...
+ (name yourbones_ppx)
+ (synopsis "A ppx companion for Yourbones")
+ (description "Some syntax extension to deal with Tezos values")
+ (depends
+  (ocaml (>= 4.14.0))
+  (dune (>= 3.0.0))
+  (ppxlib (>= 0.29.1))
+  (yourbones (= :version))
+  (mdx (and :with-test (>= 2.0.0)))
+))
```
Comment (xvw) [inline dune-project]:
```diff
-  (mdx (and :with-test (>= 2.0.0)))
+  (mdx (and :with-test (>= 2.0.0)))))
- ))
```

File: ppx/yourbones_ppx.ml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+       let concrete_value = Yourbones_common.Tez.to_int64 x in
+       fun_apply
+         "from_mutez'"
+         Ast_helper.[ Exp.constant (Const.int64 concrete_value) ]
+     | Error err ->
+       fail_with
+         {|"%s" projection into tez fail with %a|}
+         str
+         Yourbones_common.Tez.pp_error
+         err)
```
Comment (xvw) [inline ppx/yourbones_ppx.ml]: Since the code is mainly the same, WDYT about mutualizing the common behaviour?

**Review by xvw — DISMISSED** [review]
Thanks a lot! I left some small comments
And I suggest to move the `tez` part into a dedicated module (ie: `tez_literal`) and just use the registration function in `Yourbones_ppx`. It can be useful if we decide to split the library into some smaller ppx.

**Comment by mspwn** [comment]
I guess that I have adressed all of your comments. Thanks for the review.

**Review by xvw — APPROVED** [review]

**Pull request merged**